Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -320,12 +320,12 @@ def _get_self_deployed_token_auth_code(
         Returns:
             tuple: An access token, its expiry and refresh token.
         """
-        data = {
-            'client_id': self.client_id,
-            'client_secret': self.client_secret,
-            'resource': self.resource if not resource else resource,
-            'redirect_uri': self.redirect_uri
-        }
+        data = assign_params(
```

Comment:
why is this change needed?
this could lead to unexpected behavior in the other microsoft integrations

Reply:
I suggested this change since leaving the resource key with an empty value resulted in the following 400 error:
![image](https://user-images.githubusercontent.com/20818773/98089208-182fb800-1e8b-11eb-910f-7c7defcdcbd1.png)

Reply:
how come this did not come up in the past in other integrations?

Reply:
I'm not sure why this didn't come up before, but it seems that `resource` [was replaced](https://docs.microsoft.com/en-us/azure/active-directory/azuread-dev/azure-ad-endpoint-comparison#scopes-not-resources) with `scope` in `v2`.

Reply:
did we test this against the integrations that use auth code flow?

Reply:
I have just tested all those integrations that use auth code flow and seemed to be working as expected